students/shodges/lesson06/mailroom.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+import tempfile
+from pathlib import Path
+from datetime import datetime
+
+donors = {'William Henry Harrison' : [806.25, 423.10],
+          'James K. Polk' : [37.67, 127.65, 1004.29],
+          'Martin van Buren' : [126.47],
+          'Millard Fillmore' : [476.21, 2376.21],
+          'Chester A. Arthur' : [10236.91]}
+
+letter_template = """Dear {name},
+On behalf of all of us at Save the Marmots, thank you for your recent gift of ${amount:.2f}.  When it comes to ensuring marmots have loving homes, every dollar goes a long way.
+
+Your very generous gifts of ${total:.2f} will help us provide food and shelter for all of the rescued marmots, and ensure our staff have the resources to train them for placement.
+
+Warmest regards,
+
+Sean Hodges
+"""
+
+letter_whitespace = """
+
+
+
+{}
+
+
+"""
+
+###### BEGIN BLOCK OF BUSINESS FUNCTIONS (NOW WITH MORE UNIT TESTS!) ######
+
+def format_letter(donor, extra_whitespace = False):
+    try:
+        letter_values = {'name': donor, 'amount': donors[donor][-1], 'total': sum(donors[donor])}
+    except KeyError:
+        return False
+
+    if extra_whitespace == True: # I still like the extra whitespace in the user interactive mode :)
+        return (letter_whitespace.format(letter_template)).format(**letter_values)
+    else:
+        return letter_template.format(**letter_values)
+
```

Comment:
2 blank lines between functions - PEP8

Reply:
@charlesrobison Sorry!  Never responded to this.  I added the missing blank lines in [PR408](https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/408/files#diff-ef0485e7ddba92b2be36debe505ac263)